### PR TITLE
Add experiments docs

### PR DESCRIPTION
**Pull request by splatterxl**
These docs take from my personal experience and a few other sources

### CODE REVIEW

File: pages/topics/experiments.mdx [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+| b     | integer          | Bucket assigned to these resources      |
+| k     | array[snowflake] | Resources granted access to this bucket |
+
+## Endpoints
+
+<RouteHeader method="GET" url="/experiments">
+  Get Experiment Assignments
+</RouteHeader>
+
+Returns a fingerprint if not already set, and the [user experiment assignments](#user-experiments) for the fingerprint. Additionally, if `with_guild_experiments` is passed, returns the [guild experiment rollouts](#guild-experiments).
```
Comment (dolfies) [inline pages/topics/experiments.mdx:314]:
```suggestion
Returns the [user experiment assignments](#user-experiments) and optionally [guild experiment rollouts](#guild-experiments) for the user/fingerprint.

<Alert type="info">

A fingerprint will only be generated and returned if no authorization or fingerprint is provided in request headers. Fingerprint generation has a rate limit of 3 valid fingerprints per 2 minutes per IP. While fingerprints will still be returned past this rate limit, they will not be usable.

</Alert>

###### Query String Parameters

| Field | Type | Description |
| ----- | ----- | ------------- |
| with_guild_experiments? | boolean | Whether to include guild experiments in the returned data |
```

File: pages/topics/experiments.mdx [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+###### Response Structure
+
+| Field              | Type                                              | Description                                               |
+| ------------------ | ------------------------------------------------- | --------------------------------------------------------- |
+| fingerprint?       | string                                            | Returns a randomly generated [fingerprint](#fingerprints) |
```
Comment (dolfies) [inline pages/topics/experiments.mdx:320]:
```suggestion
| fingerprint?       | string                                            | A generated [fingerprint](#fingerprints) of the current date and time |
```

**Review by dolfies — CHANGES_REQUESTED** [review]
Overall looks amazing! 🥰 

One thing I didn't address in all the change requests: examples usually come after structures, and enum definitions (stuff like types) usually come after structures and examples.

File: components/Navigation.tsx [hunk 3]
```diff
@@ -437,6 +444,22 @@ export default function Navigation() {
...
+          href="/topics/experiments"
+          subLinks={
+            <Fragment>
+              <NavigationSubLink href="/topics/experiments#fingerprints">Fingerprints</NavigationSubLink>
+              <NavigationSubLink href="/topics/experiments#rollouts">Rollouts</NavigationSubLink>
+              <NavigationSubLink href="/topics/experiments#user-experiments">User Experiments</NavigationSubLink>
+              <NavigationSubLink href="/topics/experiments#guild-experiments">Guild Experiments</NavigationSubLink>
+              <NavigationSubLink href="/topics/experiments#get-experiment-assignments">
+                Get Experiment Assignments
+              </NavigationSubLink>
```
Comment (dolfies) [inline components/Navigation.tsx:457]:
```suggestion
              <NavigationSubLink href="/topics/experiments#endpoints">
                Endpoints
              </NavigationSubLink>
```

File: pages/topics/experiments.mdx [hunk 4]
```diff
@@ -0,0 +1,335 @@
+# Experiments
+
+Experiments are a form of A/B testing used by Discord in the client- and server-side of their applications to serve
+different experiences or behaviours to different users randomly and/or based on location, client version, etc.
+
+At its core, rollouts are simply a YAML value inputted in the Discord admin panel, but obviously not all of the data in
+there is required by the clients so the actual value most clients see is obfuscated and complex to decipher.
```
Comment (dolfies) [inline pages/topics/experiments.mdx:7]:
```suggestion
At their core, rollouts are simply YAML inputted into the Discord admin panel, but not all of the data
is required by the clients so the actual value most clients see is obfuscated and complex to decipher.
```

File: pages/topics/experiments.mdx [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+
+## Fingerprints
+
+Even the marketing website uses A/B tests to hook users into the app and so the app needs a unique way to identify the
+person using the website, so it resorts to using "fingerprints". Now this isn't the usual method of analysing a browser's
+unique traits, instead the [Get Experiment Rollouts](#get-experiment-rollouts) endpoint simply generates a new ID for each unauthenticated request
+which the marketing website then consumes.
```
Comment (dolfies) [inline pages/topics/experiments.mdx:14]:
```suggestion
Discord needed a way to uniquely identify people using Discord without authentication. Even the marketing website uses A/B tests to hook users into the app, and register/login flows have various experiments, so they resorted to using "fingerprints".
These aren't the usual fingerprints generated by collecting information through the browser, instead they are snowflakes generated by unauthenticated requests to [Get Experiment Rollouts](#get-experiment-rollouts). It is expected that fingerprints are sent in the `X-Fingerprint` header in all subsequent requests to the API until authentication, in order to track A/B tests and allow access to API-locked portions of experiments.
```

File: pages/topics/experiments.mdx [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+A fingerprint is comprised of a snowflake and a random hash value, presumably. It looks like this: `1084179945133187083.JQddgNMmwJPghoBtFmaH7jTmdsw`.
+
+<!-- TODO: link to auth page on this -->
+
+When signing up, the website then passes the fingerprint and the snowflake part of it is passed internally as the user's new
+ID. This means that theoretically the snowflake's creation time on a user's account represents the first time
+they ever visited the Discord homepage.
```
Comment (dolfies) [inline pages/topics/experiments.mdx:22]:
```suggestion
When registering a new account, the fingerprint is passed, and (if valid) is used as the created user's ID. This is done in order to preserve experiments across to the registered user.
Therefore, a user account's creation time theoretically represents the first time they visited Discord.
```

File: pages/topics/experiments.mdx [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+| Field                   | Type    | Description                                               |
+| ----------------------- | ------- | --------------------------------------------------------- |
+| with_guild_experiments? | boolean | Whether to include guild experiments in the returned data |
+
+###### Response Structure
+
+| Field              | Type                                              | Description                                                           |
+| ------------------ | ------------------------------------------------- | --------------------------------------------------------------------- |
+| fingerprint?       | string                                            | A generated [fingerprint](#fingerprints) of the current date and time |
+| assignments        | array[[experiment assignment](#user-experiments)] | The assignment for this fingerprint or user                           |
```
Comment (dolfies) [inline pages/topics/experiments.mdx:334]:
```suggestion
| assignments        | array[[experiment assignment](#user-experiments)] | The experiment assignments for this user or fingerprint                           |
```

File: pages/topics/experiments.mdx [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+
+All of the below objects are represented as arrays following the order in which the fields are displayed.
+
+### User Experiments
+
+User experiment data returned by the API is very limited. In contrast to the wide range of data the API provides for
+guild rollouts, the only values we can programmatically retrieve from the API are the user/[fingerprint](#fingerprints)'s assigned
+bucket for the experiment.
+
+###### User Experiment Assignment Structure
```
Comment (dolfies) [inline pages/topics/experiments.mdx:92]:
```suggestion
###### User Experiment Structure
```

**Review by dolfies — CHANGES_REQUESTED** [review]
A couple more nitpicks I missed!

File: components/Navigation.tsx (hunk 3 above)
Comment (splatterxl) [reply components/Navigation.tsx:457]: This is inconsistent with all the other pages

Comment (dolfies) [reply components/Navigation.tsx:457]: Nah it's consistent with all the changed ones, only inconsistent with the ones I haven't gotten to yet

**Review by dolfies — APPROVED** [review]

**Pull request merged**